src/all/comicskingdom/src/eu/kanade/tachiyomi/extension/all/comicskingdom/ComicsKingdom.kt

```diff
@@ -0,0 +1,285 @@
+                                addQueryParameter("order", "asc")
+                                addQueryParameter("_fields", "link,date,assets")
+                                addQueryParameter("ck_feature", mangaName)
+                                addQueryParameter("page", (idx + 1).toString())
+                            }.build().toString()
+                }
+            }.reversed()
+        }
+
+        val chapters = mutableListOf<SChapter>()
+        var pageNum = 1
+
+        var chapterData = getChapterList(mangaName, pageNum)
+        var chapterNum = 0.0F
+
+        while (chapterData != null) {
+            val list = chapterData.map {
+                chapterNum += 0.01F
+                SChapter.create().apply {
+                    Log.i("comics", it.link)
+                    chapter_number = chapterNum
+                    setUrlWithoutDomain(it.link)
+                    date_upload = dateFormat.parse(it.date).time
+                    name = it.date.substringBefore("T")
+                }
+            }
+            chapters.addAll(list)
+
+            if (list.count() < 100) {
+                Log.i("comics", "End of chapterParse")
+                break
+            }
+
+            pageNum++
+            try {
+                chapterData = getChapterList(mangaName, pageNum)
+            } catch (exception: Exception) {
+                if (chapters.isNotEmpty()) {
+                    return chapters
+                }
+            }
+        }
+
+        return chapters
+    }
+
+    private fun getChapterList(mangaName: String, page: Int): List<ComicsKingdomDto.Chapter> {
+        val url = "https://wp.$urlBase/wp-json/wp/v2/ck_comic".toHttpUrl().newBuilder().apply {
+            addQueryParameter("per_page", "" + 100) // 100 is max
+            addQueryParameter("order", "desc")
+            addQueryParameter("_fields", "link,date")
+            addQueryParameter("ck_feature", mangaName)
+            addQueryParameter("page", "" + page)
+        }.build()
+
+        val call = client.newCall(GET(url, headers)).execute()
+        val body = call.body.string()
+        call.close()
+        return json.decodeFromString<List<ComicsKingdomDto.Chapter>>(body)
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        if (chapter.url.startsWith("https://wp")) return GET(chapter.url, headers)
+        return super.pageListRequest(chapter)
+    }
+
+    override fun chapterListSelector() = throw UnsupportedOperationException()
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun pageListParse(document: Document): List<Page> {
+        if (shouldCompact()) {
+            return json.decodeFromString<List<ComicsKingdomDto.Chapter>>(document.body().text())
+                .mapIndexed { idx, chapter ->
+                    Page(idx, imageUrl = chapter.assets!!.single.url)
+                }
+        }
+
```

Comment:
override `pageListParse(response: Response)` to directly consume response body for json instead of making it a jsoup document